core/services/pipeline/orm.go

```diff
@@ -377,41 +353,42 @@ func (o *orm) ResultsForRun(ctx context.Context, runID int64) ([]Result, error)
 
 	var results []Result
 	err = postgres.GormTransaction(ctx, o.db, func(tx *gorm.DB) error {
-		var resultTaskRun TaskRun
+		var run Run
 		err = tx.Raw(`
-			SELECT * FROM pipeline_task_runs
-			WHERE pipeline_run_id = ? 
+			SELECT * FROM pipeline_runs 
+				WHERE id = ? 
 				AND finished_at IS NOT NULL
-				AND dot_id = ?
```

Comment:
the main change - we now read directly from the pipeline run

Reply:
Finally 🙌 

This should be a more efficient lookup as well.